storage.lua

```diff
@@ -246,6 +247,57 @@ function mail.extractMaillists(receivers_string, maillists_owner)
 	return receivers
 end
 
+function mail.get_setting_default_value(setting_name)
+	local default_values = {
+		chat_notifications = true,
+		onjoin_notifications = true,
+		hud_notifications = true,
+		unreadcolorenable = true,
+		cccolorenable = true,
+		defaultsortfield = 3,
+		defaultsortdirection = 1,
+	}
+	return default_values[setting_name]
+end
+
+function mail.get_setting(playername, setting_name)
+	local entry = mail.get_storage_entry(playername)
+	local setting = nil
+	for _, existing_setting in ipairs(entry.settings) do
+		if existing_setting.name == setting_name then
+			setting = existing_setting.value
+			break
+		end
+	end
+	if setting == nil then setting = mail.get_setting_default_value(setting_name) end
+	return setting
+end
+
+-- add or update a setting
+function mail.set_setting(playername, setting)
+	local entry = mail.get_storage_entry(playername)
+	local existing_updated = false
+	for i, existing_setting in ipairs(entry.settings) do
+		if existing_setting.name == setting.name then
+			-- update
+			entry.settings[i] = setting
+			existing_updated = true
+			break
+		end
+	end
+	if not existing_updated then
+		-- insert
+		table.insert(entry.settings, setting)
+	end
+	mail.set_storage_entry(playername, entry)
+end
```

Comment:
Would it be better to have more restricted interface here:
```diff
-function mail.set_setting(playername, setting)
+function mail.set_setting(playername, key, value)
        local entry = mail.get_storage_entry(playername)
-       local existing_updated = false
        for i, existing_setting in ipairs(entry.settings) do
-               if existing_setting.name == setting.name then
+               if existing_setting.name == key then
                        -- update
-                       entry.settings[i] = setting
-                       existing_updated = true
-                       break
+                       existing_setting.value = value
+                       mail.set_storage_entry(playername, entry)
+                       return
                end
        end
-       if not existing_updated then
-               -- insert
-               table.insert(entry.settings, setting)
-       end
+       -- insert
+       table.insert(entry.settings, { name = key, value = value })
        mail.set_storage_entry(playername, entry)
 end
```

Of course doing that would also mean that it should be updated where set_settings is called.

<details><summary>Same as GitHub suggestion, I just find it harder to read:</summary>

```suggestion
function mail.set_setting(playername, key, value)
	local entry = mail.get_storage_entry(playername)
	for i, existing_setting in ipairs(entry.settings) do
		if existing_setting.name == key then
			-- update
			existing_setting.value = value
			mail.set_storage_entry(playername, entry)
			return
		end
	end
	-- insert
	table.insert(entry.settings, { name = key, value = value })
	mail.set_storage_entry(playername, entry)
end
```

</details>